### PR TITLE
Mypy fixes

### PR DESCRIPTION
**Pull request by BengtRydberg**
for a few files, primarily in the data_processing directory. This PR seems to resolve all mypy errors of modules in this directory.

### CODE REVIEW

File: scanomatic/data_processing/norm.py [hunk 1]
```diff
@@ -89,8 +89,8 @@ def infer_offset(arr):
 
 
 def get_downsampled_plates(
-    data,
-    subsampling: Literal['TL', 'TR', 'BL', 'BR'] = "BR",
+    data: np.ndarray,
+    subsampling: Union[List[np.ndarray], Literal['TL', 'TR', 'BL', 'BR']] = "BR",
```
Comment (local-minimum) [inline scanomatic/data_processing/norm.py]: in python 3.9 we don't need `List`, `Tuple`, `Dict` anymore, we can use `list`, `tuple` `dict` directly. Various places below

**Review by local-minimum — CHANGES_REQUESTED** [review]

File: scanomatic/generics/maths.py [hunk 2]
```diff
@@ -14,23 +14,25 @@ def iqr_mean(data: np.ndarray, *args, **kwargs) -> Optional[np.ndarray]:
...
-        data = np.ma.masked_invalid(data)
-
-    data = data[data.mask == False]  # noqa: E712
+    data = get_finite_data(data)
     threshold = int(np.floor(data.size * 0.25))
     data.sort()
     return data[threshold], data[-threshold]
+
+
+def get_finite_data(
```
Comment (skymandr) [inline scanomatic/generics/maths.py:31]: Maybe add a test?

File: scanomatic/data_processing/norm.py (hunk 1 above)
Comment (BengtRydberg) [reply scanomatic/data_processing/norm.py]: ok, fixed.

File: scanomatic/generics/maths.py (hunk 2 above)
Comment (BengtRydberg) [reply scanomatic/generics/maths.py:31]: added unit test for this function.

File: scanomatic/data_processing/data_bridge.py [hunk 3]
```diff
@@ -36,7 +37,9 @@ def _create_array_representation(self, **kwargs):
             self.update_source = self._update_to_array
 
         elif isinstance(self._source, dict):
-            plates = [[]] * len(self._source)  # Creates plates and 1D pos
+            plates: List[
+                List[List[List[float]]]
+            ] = [[]] * len(self._source)  # Creates plates and 1D pos
```
[the remark this answers is not in the document]
Comment (BengtRydberg) [reply scanomatic/data_processing/data_bridge.py]: not used elsewhere in the module so I do not think it is of much help to declare it.

File: scanomatic/data_processing/phases/segmentation.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import operator
 from enum import Enum
-from typing import Optional, Union
+from typing import Iterator, Optional, Union, cast
```
Comment (local-minimum) [inline scanomatic/data_processing/phases/segmentation.py:3]: Iterator, Sequence etc are deprecated in `typing` and should be imported from `collections.abc`. It's mostly `Optional`, `TypeVar`, `Union`, `Any` and `cast` that should come from `typing`.

File: scanomatic/data_processing/subplates.py [hunk 5]
```diff
@@ -1,8 +1,12 @@
+from typing import Optional, Union
 import numpy as np
 
 from . import mock_numpy_interface
 
 
+TUPLE_TYPE = Union[tuple[int, int], tuple[int, int, int]]
```
Comment (local-minimum) [inline scanomatic/data_processing/subplates.py:7]: maybe `PlateShape` or `Plate` is better type name?

Comment (BengtRydberg) [reply scanomatic/data_processing/subplates.py]: agree. changed to PlateShape.

File: scanomatic/data_processing/phases/segmentation.py (hunk 4 above)
Comment (BengtRydberg) [reply scanomatic/data_processing/phases/segmentation.py]: thanks for the info. fixed.

**Review by local-minimum — APPROVED** [review]

**Pull request merged**